internal/client/gitea.go

```diff
@@ -0,0 +1,161 @@
+package client
+
+import (
+	"os"
+	"strconv"
+
+	"code.gitea.io/gitea/modules/structs"
+	"code.gitea.io/sdk/gitea"
+	"github.com/apex/log"
+	"github.com/goreleaser/goreleaser/internal/artifact"
+	"github.com/goreleaser/goreleaser/internal/tmpl"
+	"github.com/goreleaser/goreleaser/pkg/config"
+	"github.com/goreleaser/goreleaser/pkg/context"
+)
+
+type giteaClient struct {
+	client *gitea.Client
+}
+
+// NewGitea returns a gitea client implementation
+func NewGitea(ctx *context.Context) (Client, error) {
+	token := ctx.Token
+	client := gitea.NewClient(ctx.Config.Release.Gitea.InstanceURL, token)
+	return &giteaClient{client: client}, nil
+}
+
+// CreateFile creates a file in the repository at a given path
+// or updates the file if it exists
+func (c *giteaClient) CreateFile(
+	ctx *context.Context,
+	commitAuthor config.CommitAuthor,
+	repo config.Repo,
+	content []byte,
+	path,
+	message string,
+) error {
+	// doesn't have to be implemented for Gitea
+	return nil
+}
+
+func (c *giteaClient) createRelease(ctx *context.Context, title, body string) (*gitea.Release, error) {
+	releaseConfig := ctx.Config.Release
+	owner := releaseConfig.Gitea.Owner
+	repoName := releaseConfig.Gitea.Name
+	tag := ctx.Git.CurrentTag
+
+	opts := structs.CreateReleaseOption{
+		TagName:      tag,
+		Target:       ctx.Git.Commit,
+		Title:        title,
+		Note:         body,
+		IsDraft:      releaseConfig.Draft,
+		IsPrerelease: ctx.PreRelease,
+	}
+	release, err := c.client.CreateRelease(owner, repoName, opts)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"err": err.Error(),
+		}).Debug("error creating Gitea release")
+		return nil, err
+	}
+	log.WithField("id", release.ID).Info("Gitea release created")
+	return release, nil
+}
+
+func (c *giteaClient) getExistingRelease(owner, repoName, tagName string) (*gitea.Release, error) {
+	releases, err := c.client.ListReleases(owner, repoName)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, release := range releases {
```

Comment:
could become quite expensive for a lot of repositories. Probably using https://gitea.com/gitea/go-sdk/src/branch/master/gitea/release.go#L28 suites better

Reply:
You need a release id for that. Where would you get it from?

Reply:
Ah I see, you're right :) Well [repoListReleases](https://try.gitea.io/api/swagger#/repository/repoListReleases) also has pagination, but I assume adding no or a negative value gives you all releases right?

Reply:
I will look into it!

Reply:
Well, the SDK doesn't allow for pagination https://godoc.org/code.gitea.io/sdk/gitea#Client.ListReleases

Reply:
okay I see, the sdk would need some updates. Then we keep it the way you implemented it